src/main/java/com/exactpro/th2/fix/client/Main.java

```diff
@@ -274,7 +288,8 @@ public void setSessionSettings(List<FixBean> sessionSettings) throws IncorrectDa
                             "Repeating of session alias: \"" + sessionAlias + "\" or sessionID: \"" + sessionID + "\"");
                 }
             }
-            this.sessionSettings = sessionSettings;
+            this.sessionIDsByAliases = Map.copyOf(sessionIDsByAliases);
+            this.sessionSettings = List.copyOf(sessionSettings);
```

Comment:
Why do we copy them? Wrapping using `Collections.unmodifiable*` would be enough and more lightweight